File: java-streams/examples/java/src/main/java/streams/Find.java
package streams;

import java.util.List;
import java.util.Optional;
import java.util.stream.Stream;

public class Find {

    void examples(List<Person> persons) {

        Stream<Person> stream = persons.stream();

        Optional<Person> firstBlackadder = stream
                .filter(person -> person.getLastName().equals("Blackadder"))
                .findFirst();

        Optional<Person> whicheverBlackadder = stream
                .filter(person -> person.getLastName().equals("Blackadder"))
                .findAny();

        boolean anyMembersOfBlackadderFamily = stream
                .anyMatch(person -> person.getLastName().equals("Blackadder"));

        boolean justBlackadderFamily = stream
                .allMatch(person -> person.getLastName().equals("Blackadder"));
    }
}

File: java-streams/examples/kotlin/src/main/kotlin/streams/Find.kt
package streams

class Find {

    internal fun examples(persons: List<Person>) {

        val firstBlackadder =
                persons.first { it.lastName == "Blackadder" }

        val whicheverBlackadder =
                persons.find { it.lastName == "Blackadder" }

        val anyMembersOfBlackadderFamily =
                persons.any { it.lastName == "Blackadder" }

        val justBlackadderFamily =
                persons.all { it.lastName == "Blackadder" }
    }
}
